source/common/router/config_impl.h

```diff
@@ -356,7 +310,11 @@ class RouteEntryImplBase : public RouteEntry,
   std::chrono::milliseconds timeout() const override { return timeout_; }
   const VirtualHost& virtualHost() const override { return vhost_; }
   bool autoHostRewrite() const override { return auto_host_rewrite_; }
-  bool useWebSocket() const override { return use_websocket_; }
+  bool useWebSocket() const override { return websocket_config_ != nullptr; }
+  Http::WebSocketProxyPtr createWebSocketProxy(
+      Http::HeaderMap& request_headers, const RequestInfo::RequestInfo& request_info,
+      Http::WebSocketProxyCallbacks& callbacks, Upstream::ClusterManager& cluster_manager,
```

Comment:
Somehow this doesn’t feel like the best way to go about this. ESP because the routing table is now creating new objects / allocating stuff. Till date we have only been returning references to pre allocated stuff and the table has remained a read only thingy